registry/cache/repocachemanager.go

```diff
@@ -24,25 +24,35 @@ type imageToUpdate struct {
 
 // repoCacheManager handles cache operations for a container image repository
 type repoCacheManager struct {
-	now         time.Time
-	repoID      image.Name
-	burst       int
-	trace       bool
-	logger      log.Logger
-	cacheClient Client
+	now           time.Time
+	repoID        image.Name
+	client        registry.Client
+	clientTimeout time.Duration
+	burst         int
+	trace         bool
+	logger        log.Logger
+	cacheClient   Client
 	sync.Mutex
 }
 
-func newRepoCacheManager(now time.Time, repoId image.Name, burst int, trace bool, logger log.Logger,
-	cacheClient Client) *repoCacheManager {
-	return &repoCacheManager{
-		now:         now,
-		repoID:      repoId,
-		burst:       burst,
-		trace:       trace,
-		logger:      logger,
-		cacheClient: cacheClient,
+func newRepoCacheManager(now time.Time,
+	repoID image.Name, clientFactory registry.ClientFactory, creds registry.Credentials, repoClientTimeout time.Duration,
+	burst int, trace bool, logger log.Logger, cacheClient Client) (*repoCacheManager, error) {
```

Comment:
I don't think the amount of parameters (which comes from the previous refactoring I made) is reasonable at all.

I tried a few alternatives and they were worse. Any suggestions?

Reply:
Some people (including me occasionally) wrap optional values in a config struct, which is passed by value.

Reply:
Uhm, only one (or maybe two) are really optional.